torch/csrc/jit/codegen/cuda/lower_unroll.cpp

```diff
@@ -168,6 +182,7 @@ kir::ForLoop* handleMisalignedVectorization(
   auto in_tv = vec_expr->inputs().front()->as<kir::TensorView>();
   kir::TensorView* vec_tv =
       (out_tv->memoryType() != MemoryType::Global) ? out_tv : in_tv;
+  kir::TensorView* non_vec_tv = (vec_tv == out_tv) ? in_tv : vec_tv;
```

Comment:
Maybe just check `out_tv->memoryType() == MemoryType::Global`. It seems more clear to me in hindsight.